brage-import/src/main/java/no/sikt/nva/brage/migration/merger/PreMergeValidator.java

```diff
@@ -0,0 +1,34 @@
+package no.sikt.nva.brage.migration.merger;
+
+import no.unit.nva.model.Publication;
+import no.unit.nva.model.contexttypes.Degree;
+import nva.commons.core.JacocoGenerated;
+
+public final class PreMergeValidator {
+
+    public static final String HANDLE_ALREADY_EXIST_ERROR_MESSAGE = "Cannot merge student degree with nva publication"
+                                                                    + " with handle in additional identifiers";
+
+    @JacocoGenerated
+    private PreMergeValidator() {
+
+    }
+
+    public static void validate(Publication existingPublication) {
+        if (isDegree(existingPublication)) {
+            if (hasHandlePresentAlready(existingPublication)) {
+                throw new DegreeMergingException(HANDLE_ALREADY_EXIST_ERROR_MESSAGE);
+            }
```

Comment:
Nice, maybe we could persist cristinId or handle of existing publication in error report body?